### DISCUSSION TITLE
'stylex.create' should never be called at runtime. It should be compiled away by '@stylexjs/babel-plugin'

### DISCUSSION THREAD

**Discussion by yug-delemont** [facebook/stylex#397] [Q&A]
![image](https://github.com/facebook/stylex/assets/154723133/45f94608-b5f1-4ec0-b2b7-cada4d2e31a9)

`'stylex.create' should never be called at runtime. It should be compiled away by '@stylexjs/babel-plugin'

I face the same error in my react app I don't understand, why is not working in my app can you help me find a solution to this?

`import * as stylex from '@stylexjs/stylex';

const styles = stylex.create({
base: {
fontSize: 16,
lineHeight: 1.5,
color: 'rgb(60,60,60)',
},
highlighted: {
color: 'rebeccapurple',
},
});

 <div {...stylex.props(styles.base, styles.highlighted)}>
          HELLO WORLD
        </div>`

**Comment by nmn** [comment]
Depends on your project set up. You haven't set up the Babel plugin correctly. Will need more context to help.

**Reply by yug-delemont** [reply]
Thanks for replying, I hope the below information helps you to understand which error I am facing.

This is my App.tsx
---
![image](https://github.com/facebook/stylex/assets/154723133/778df26d-1be8-4b46-9aae-7eb3c0596e8c)

This is my package.json
---
![image](https://github.com/facebook/stylex/assets/154723133/43496106-b358-4afc-87c0-74077e4f5ada)

New Error
---
![image](https://github.com/facebook/stylex/assets/154723133/f6743437-1968-48d7-8112-7124d4d8a0ba)

Babel Config
---
![image](https://github.com/facebook/stylex/assets/154723133/499f1ff1-f6db-461a-8dce-01302ffec021)

As per https://stylexjs.com i follow defining and using stylex.js but it's not working in my app also I watched many YouTube videos but it's still not working.

I follow babel as par https://stylexjs.com/docs/learn/installation/ 

can i need to more config such as webpack and esbuild?

**Reply by yug-delemont** [reply]
You can see error on this pic.

![image](https://github.com/facebook/stylex/assets/154723133/3793c6af-79a8-42c1-ac83-9140f03a234a)

**Reply by edamascus** [reply]
It seems that you have created your app using `create-react-app` and you are having the same problem discussed in issue #289.

Until a plugin for `create-react-app` is available. You may need to eject and use the plugin which is available for [Webpack](https://stylexjs.com/docs/learn/installation/#production).

Another option would be to setup your app using a different bundler like [Vite ](https://vitejs.dev/) which has a [plugin](https://github.com/HorusGoul/vite-plugin-stylex) for StyleX that is easy to setup and works well with React CSR apps.
